framework/src/application/schema/merge_deep.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+interface KVPair {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	[key: string]: any;
+}
+
+const isObject = (item: unknown): item is KVPair =>
+	typeof item === 'object' && item !== null && !Array.isArray(item);
+
+export const mergeDeep = (dest: KVPair, src: KVPair): void => {
```

Comment:
I would suggest using `lodash.merge` package. We already using similar for `lodash.cloneDeep`. I think a merge is not that simple otherwise lodash would not have hundred of lines for this.

Reply:
I would prefer not to include the extra package if the implementation is as simple like this